src/show.jl

```diff
@@ -16,35 +38,115 @@ function _string(x::MagOrRef)
     return read(out_rd, String)
 end
 
-function Base.show(io::IO, x::MagOrRef)
-    if isdefined(Main, :IJulia) && Main.IJulia.inited
-        print(io, Float64(x))
-    else
-        print(io, _string(x))
-    end
+function Base.string(
+    x::MagOrRef;
+    digits::Integer = digits_prec(30),
+    remove_trailing_zeros::Bool = true,
+)
+    cstr = ccall(@libflint(arf_get_str), Ptr{UInt8}, (Ref{arf_struct}, Int), Arf(x), digits)
+    str = unsafe_string(cstr)
+    ccall(@libflint(flint_free), Nothing, (Ptr{UInt8},), cstr)
+
+    return remove_trailing_zeros ? _remove_trailing_zeros(str) : str
+end
```

Comment:
we could just replace this with
```suggestion
function Base.string(
    x::MagOrRef;
    digits::Integer = digits_prec(30),
    remove_trailing_zeros::Bool = true,
)
    return string(Arf(x); digits, remove_trailing_zeros)

end
```

I actually didn't know that one can pass kwargs without explicitly writing them! when was it introduced?!